plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/GitHook.kt

```diff
@@ -151,19 +150,19 @@ open class KtlintInstallGitHookTask @Inject constructor(
     @get:Input
     internal val hookName: Property<String> = objectFactory.property(String::class.java)
 
-    @get:InputDirectory
-    internal val projectDir: DirectoryProperty = objectFactory.directoryProperty().apply {
-        set(projectLayout.projectDirectory)
+    @get:Input
+    internal val projectDir: Property<String> = objectFactory.property(String::class.java).apply {
+        set(projectLayout.projectDirectory.asFile.absolutePath)
     }
 
-    @get:InputDirectory
-    internal val rootDirectory: DirectoryProperty = objectFactory.directoryProperty().apply {
-        set(project.rootDir)
+    @get:Input
```

Comment:
this is the correct way to make an input depend on the _path_ and not the _contents_ :+1: